spec/services/container_deployment_service_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'recursive-open-struct'
+
+describe ContainerDeploymentService do
+  before do
+    %w(amazon openstack google azure redhat vmware).each do |p|
+      network = FactoryGirl.create(:network, :ipaddress => "127.0.0.1")
+      hardware = FactoryGirl.create(:hardware)
+      hardware.networks << network
+      vm = FactoryGirl.create("vm_#{p}".to_sym, :hardware => hardware)
+      if (p.include? "amazon") || (p.include? "redhat")
```

Comment:
> if ['amazon', 'redhat'].include?(p) ?

these are the only two providers we currently support for provision new vms in the openshift deployment. 
I can add a comment here, but looking at containerDeploymentService makes it clear